crates/ty_python_semantic/src/types.rs

```diff
@@ -4609,14 +4640,19 @@ impl<'db> Type<'db> {
                         )
                     })
                     .unwrap_or(instance_ty);
-                Ok(specialized)
+
+                // Return the __new__ type if available, otherwise return instance_ty
```

Comment:
This comment looks like it belongs a few lines down; there's no `instance_ty` here